### PR TITLE
Incremental: do not call Manage_subdevs() from IncrementalRemove()

### PR DESCRIPTION
**Pull request by mtkaczyk**
This is a first part of HOT_REMOVE_DISK/SET_FAULTY rework. The goal is to not execute HOT_REMOVE_DISK ioctl on `mdadm --remove` path with container.

This PR changes following:
- Manage_subdevs will not execute HOT_REMOVE_IOCTL with "faulty" (not needed because sysfs way is same reliable).
- Incremental_remove will implement remove logic directly.

### CODE REVIEW

**Review by bkucman — CHANGES_REQUESTED** [review]
Hi, 
Overall I agree with the change, I have a few minor questions or corrections.

Typo in commit desc ee89efad3181fb9adbf39f52d9fdd0d2e263349a
dediated -> dedicated

Unnecessary dots in commit titles

**Comment by mtkaczyk** [comment]
@bkucman
thanks for review, For now moving again to draft, I opened new PR #46 - it is a preparation for this change and needs to come first.

**Comment by mtkaczyk** [comment]
Part of this (support for devnode in mdadm -If) is moved to #87
After merging #87 I will get back to this PR.

File: Incremental.c
Comment (mtkaczyk) [reply Incremental.c:1739]: I don't think it is important, we have to try remove the device anyway.

File: Incremental.c
Comment (mtkaczyk) [reply Incremental.c]: I think it part of it is done.

File: Incremental.c
Comment (mtkaczyk) [reply Incremental.c:1738]: There are more places like that, so not in this patchset, sorry :(

File: Incremental.c
Comment (mtkaczyk) [reply Incremental.c]: unfortunately not.

**Comment by mtkaczyk** [comment]
I tested it functionally on my own with native and IMSM. No regression detected but it was minimal testing.
I will merge it after the release. I'm waiting for Kuai patches for bitmap right now and then I will make a release.

You can see testing result here:
https://github.com/md-raid-utilities/mdadm/actions/runs/11969166801/job/33369355291?pr=31
I see no new failure but I will look closer later.

**Comment by mtkaczyk** [comment]
@ncroxon, comparing to Nightly execution on the latest main. I see `01r5fail` fixed probably by this PR.

`05r1-internalbitmap-v1c` is broken but this functionality and test will be removed soon.

**Pull request merged**
